geoopt/manifolds/scaled.py

```diff
@@ -0,0 +1,182 @@
+    >>> import geoopt, torch, numpy as np
+    >>> sphere = geoopt.Sphere()
+    >>> radius_2_sphere = Scaled(sphere, 2)
+    >>> p1 = torch.tensor([-1., 0.])
+    >>> p2 = torch.tensor([0., 1.])
+    >>> np.testing.assert_allclose(sphere.dist(p1, p2), np.pi / 2)
+    >>> np.testing.assert_allclose(radius_2_sphere.dist(p1, p2), np.pi)
+    """
+
+    def __init__(self, manifold: Manifold, scale=1.0, learnable=False):
+        super().__init__()
+        self.base = manifold
+        scale = torch.as_tensor(scale, dtype=torch.get_default_dtype())
+        scale = scale.requires_grad_(False)
+        if not learnable:
+            self.register_buffer("_scale", scale)
+            self.register_buffer("_log_scale", None)
+        else:
+            self.register_buffer("_scale", None)
+            self.register_parameter("_log_scale", torch.nn.Parameter(scale.log()))
+        # do not rebuild scaled functions very frequently, save them
+
+        for method, scaling_info in self.base.__scaling__.items():
+            # register rescaled functions as bound methods of this particular instance
+            unbound_method = getattr(self.base, method).__func__  # unbound method
+            self.__setattr__(
+                method, types.MethodType(rescale(unbound_method, scaling_info), self)
+            )
+
+    @property
+    def scale(self):
+        if self._scale is None:
+            return self._log_scale.exp()
+        else:
+            return self._scale
+
+    @property
+    def log_scale(self):
+        if self._log_scale is None:
+            return self._scale.log()
+        else:
+            return self._log_scale
+
+    # propagate all important stuff
+    reversible = property(lambda self: self.base.reversible)
+    ndim = property(lambda self: self.base.ndim)
+    name = "Scaled"
+    __scaling__ = property(lambda self: self.base.__scaling__)
+
+    # Make AbstractMeta happy, to be fixed in __init__
+    retr = NotImplemented
+    expmap = NotImplemented
+
+    def __getattr__(self, item):
+        try:
+            return super().__getattr__(item)
+        except AttributeError as original:
+            try:
+                # propagate only public methods and attributes, ignore buffers, parameters, etc
+                if isinstance(self.base, Scaled) and item in self._base_attributes:
+                    return self.base.__getattr__(item)
+                else:
+                    return self.base.__getattribute__(item)
+            except AttributeError as e:
+                raise original from e
+
+    @property
+    def _base_attributes(self):
+        if isinstance(self.base, Scaled):
+            return self.base._base_attributes
+        else:
+            base_attribures = set(dir(self.base.__class__))
+            base_attribures |= set(self.base.__dict__.keys())
+            return base_attribures
+
+    def __dir__(self):
+        return list(set(super().__dir__()) | self.base_attribures)
+
+    def __repr__(self):
```

Comment:
`__str__` would be better choice here.

Reply:
Repr is considered consistent with pytorch api